.infrastructure/helm-chart/todoapp/templates/app/hpa.yml

```diff
@@ -0,0 +1,25 @@
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Chart.Name }}-hpa
+  namespace: {{ .Values.common.namespace }}
```

Comment:
Ensure that `.Values.common.namespace` is defined in your `values.yaml` file. If it's not set, the HPA might not be created in the intended namespace.